pkg/functionconfig/types.go

```diff
@@ -34,18 +34,30 @@ type DataBinding struct {
 	Attributes map[string]string `json:"attributes,omitempty"`
 }
 
+// Partition is a partition information
+type Partition struct {
+	ID         string  `json:"id"`
+	Checkpoint *string `json:"checkpoint,omitempty"`
```

Comment:
should be trigger.Checkpoint

Reply:
This will create import cycle. I'll do a `type Checkpoint *string` here and change trigger.